spikeinterface/sortingcomponents/waveforms/temporal_pca.py

```diff
@@ -0,0 +1,231 @@
+            we, n_components=n_components, mode="by_channel_global", sparsity=sparsity, whiten=whiten
+        )
+
+        pca_model = pc.get_pca_model()
+        waveform_time_interval_dict = {
+            "ms_before": ms_before,
+            "ms_after": ms_after,
+            "sampling_frequency": recording.get_sampling_frequency(),
+        }
+
+        # Load the model and the time interval dict from the model_folder
+        if model_folder_path is not None and Path(model_folder_path).is_dir():
+            model_path = Path(model_folder_path) / "pca_model.pkl"
+            with open(model_path, "wb") as f:
+                pickle.dump(pca_model, f)
+            model_time_interval_path = Path(model_folder_path) / "time_interval.json"
+            with open(model_time_interval_path, "w") as f:
+                json.dump(waveform_time_interval_dict, f)
+
+        return model_folder_path
+
+
+TemporalPCBaseNode.fit.__doc__ = TemporalPCBaseNode.fit.__doc__.format(_shared_job_kwargs_doc)
+
+
+class TemporalPCAProjection(TemporalPCBaseNode):
+    """
+    A step that performs a PCA projection on the waveforms extracted by a peak_detection function.
+
+    This class can work in two ways:
+    1) The class can receive the path of a previously trained model with the `model_folder_path` argument.
+    2) The class can be trained with the fit method after instantiating the class. The model will be stored in
+    `model_folder_path`.
+
+
+    Parameters
+    ----------
+    recording : BaseRecording
+        The recording object.
+    parents: list
+        The parent nodes of this node. This should contain a mechanism to extract waveforms.
+    model_folder_path : str, Path
+        The path to the folder containing the pca model and the training metadata.
+    return_output: bool, optional, true by default
+        use false to suppress the output of this node in the pipeline
+
+    """
+
+    def __init__(
+        self, recording: BaseRecording, parents: list[PipelineNode], model_folder_path: str, return_ouput=True
+    ):
+        TemporalPCBaseNode.__init__(
+            self, recording=recording, parents=parents, return_ouput=return_ouput, model_folder_path=model_folder_path
+        )
+
+    def compute(self, traces: np.ndarray, peaks: np.ndarray, waveforms: np.ndarray) -> np.ndarray:
+        """
+        Projects the waveforms using the PCA model trained in the fit method or loaded from the model_folder_path.
+
+        Parameters
+        ----------
+        traces : np.ndarray
+            The traces of the recording.
+        peaks : np.ndarray
+            The peaks resulting from a peak_detection step.
+        waveforms : np.ndarray
+            Waveforms extracted from the recording using a WavefomExtractor node.
+
+        Returns
+        -------
+        np.ndarray
+            The projected waveforms.
+
+        """
+
+        num_waveforms, num_samples, num_channels = waveforms.shape
```

Comment:
```suggestion
        num_channels = waveforms.shape[2]
```
Not using the other two